newIDE/app/src/AnnouncementsFeed/index.js

```diff
@@ -56,13 +58,58 @@ export const AnnouncementsFeed = ({
 
   if (!displayedAnnouncements.length) return null;
 
+  const getAnnouncementContent = (
+    i18n: I18nType,
+    announcement: Announcement
+  ): {|
+    title: string,
+    message: string,
+    isMarkdownImageOnly?: boolean,
+  |} => {
+    const title = selectMessageByLocale(i18n, announcement.titleByLocale);
+    const message = selectMessageByLocale(
+      i18n,
+      announcement.markdownMessageByLocale
+    );
+
+    if (!title) {
+      const markdownClickableImageRegex = /\[!\[(?<alt>[^\][]*)\]\((?<imageSource>[^\s]*)\)\]\((?<linkHref>[^\s]*)\)/;
+      let matches = message.match(markdownClickableImageRegex);
+      if (
+        matches &&
+        matches.groups &&
+        matches.groups.alt &&
+        matches.groups.imageSource &&
+        matches.groups.linkHref
+      ) {
+        return { title, message, isMarkdownImageOnly: true };
+      }
+      const markdownImageRegex = /!\[(?<alt>[^\][]*)\]\((?<imageSource>[^\s]*)\)/;
+      matches = message.match(markdownImageRegex);
+      if (
+        matches &&
+        matches.groups &&
+        matches.groups.alt &&
+        matches.groups.imageSource
+      ) {
+        return { title, message, isMarkdownImageOnly: true };
+      }
+    }
+    return { title, message };
+  };
+
   return (
     <I18n>
       {({ i18n }) => (
         <Line noMargin={!addMargins}>
           <ColumnStackLayout noMargin={!addMargins} expand>
             {displayedAnnouncements.map(announcement => {
               const { buttonLabelByLocale, buttonUrl } = announcement;
+              const {
+                title,
+                message,
+                isMarkdownImageOnly,
+              } = getAnnouncementContent(i18n, announcement);
```

Comment:
Not sure if worth it, but you could have useMemo the result of this by mapping before the render.
Maybe this is premature optimization though

Reply:
I think this optimization will be useful if we extract the rendered announcement into a new component